### PR TITLE
CI/CD Upgrade.

### PR DESCRIPTION
**Pull request by pwei1018**
*Issue #:* https://github.com/bcgov/entity/issues/7578###

*Description of changes:*
1. Deprecate Jenkins pipeline;
2. Handle test and prod CD flow by Makefile and GHA;
3. Enable Github environments for each repo;
4. Using multiple steps check in GHA CI flow;

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -18,28 +32,47 @@ jobs:
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install dependencies
+        run: |
+          make setup
+      - name: Test with pytest
```
Comment (severinbeauvais) [inline .github/workflows/ci.yml:59]: There's no pytest for UI code :)

It uses Jest.

File: .github/workflows/cd.yml [hunk 2]
```diff
@@ -7,31 +7,51 @@ on:
...
         required: true
         default: "dev"
 
 jobs:
+  push:
+    runs-on: ubuntu-20.04
+
+    if: github.event_name == 'push' && github.repository == 'bcgov/business-create-ui'
+    environment:
+      name: "dev"
```
Comment (severinbeauvais) [inline .github/workflows/cd.yml:20]: So how does this work? Does it always do a push, which deploys "main branch" to dev environment?

Is it possible to simply "dispatch" dev -> test or test -> prod?

File: Makefile [hunk 3]
```diff
@@ -52,7 +63,7 @@ push: #build ## Push the docker container to the registry & tag latest
 
 VAULTS=`cat devops/vaults.json`
 update-env: ## Update env from 1pass
-	oc -n "$(OPS_REPOSITORY)-tools" exec dc/bcros-cli -- ./scripts/1pass.sh \
+	oc -n "$(OPS_REPOSITORY)-$(TAG_NAME)" exec "dc/vault-service-$(TAG_NAME)" -- ./scripts/1pass.sh \
```
Comment (severinbeauvais) [inline Makefile:66]: Does this get done on every deploy (from GHA CD), or only for main -> dev ?

Comment (pwei1018) [reply Makefile:66]: Yes, it will update the envrs from 1pass for all envs.

File: .github/workflows/cd.yml (hunk 2 above)
Comment (pwei1018) [reply .github/workflows/cd.yml:20]: Push commit can be only deployed to dev. Test and prod requires dispatch action to deploy. Also need reviewers to approve it. I may explain it in Tech review next time.

Comment (severinbeauvais) [reply .github/workflows/cd.yml:20]: Sounds good.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (pwei1018) [reply .github/workflows/ci.yml]: Fixed

**Review by severinbeauvais — APPROVED** [review]

**Pull request merged**